lumibot/brokers/broker.py

```diff
@@ -56,12 +61,299 @@ def __init__(self, name="", connect_stream=True):
             if self.stream is not None:
                 self._launch_stream()
 
+    # =================================================================================
+    # ================================ Required Implementations========================
+    # =========Order Handling=======================
+    @abstractmethod
+    def cancel_order(self, order: Order):
+        """Cancel an order at the broker"""
+        pass
+
+    @abstractmethod
+    def _submit_order(self, order: Order):
+        """Submit an order to the broker"""
+        pass
+
+    # =========Account functions=======================
+    @abstractmethod
+    def _get_balances_at_broker(self, quote_asset: Asset) -> float:
+        """
+        Get the actual cash balance at the broker.
+        Parameters
+        ----------
+        quote_asset : Asset
+            The quote asset to get the balance of.
+
+        Returns
+        -------
+        float
+            The balance of the quote asset at the broker.
+        """
+        pass
+
+    @abstractmethod
+    def get_historical_account_value(self):
+        """
+        Get the historical account value of the account.
+        TODO: Fill out the docstring with more information.
+        """
+        pass
+
+    # =========Stream functions=======================
+    @abstractmethod
+    def _get_stream_object(self):
+        """get the broker stream connection"""
+        pass
+
+    @abstractmethod
+    def _register_stream_events(self):
+        """Register the function on_trade_event
+        to be executed on each trade_update event"""
+        pass
+
+    @abstractmethod
+    def _run_stream(self):
+        """Run the stream"""
+        pass
+
+    # =========Broker Positions=======================
+    @abstractmethod
```

Comment:
We can make a new Task/Issue for this in the Project Plan.  I think we need to put out heads together a bit on this because I'm not entirely clear on the structure of the data being produced.  Once this separation is fully complete I think it will be much easier to return to this and make it work as we want.

Reply:
makes sense, agreed